app/livechat/server/api/lib/livechat.js

```diff
@@ -71,9 +71,9 @@ export function findOpenRoom(token, departmentId) {
 	}
 
 	if (room) {
-		Livechat.addTypingListener(room._id, (username, typing) => {
+		Livechat.addTypingListener(room._id, _.debounce((username, typing) => {
 			Apps.triggerEvent(AppEvents.IRoomUserTyping, { roomId: room._id, username, typing });
-		});
+		}, 4000, true));
```

Comment:
Hi @Shailesh351, I reviewed the PR, but the delay between the visitor starting to type and the indicator actually been shown to the agent is a little too much. Can we decrease that delay by a couple of seconds?

Reply:
Hey @PrajvalRaval I tried some values but the number of triggers was high even if user is typing continuously with normal speed. 
In the LiveChat widget https://github.com/RocketChat/Rocket.Chat.Livechat/blob/48ae3dc366056c9985849719beed90a3fb87d50f/src/routes/Chat/container.js#L99 we have **4500 ms**.
May be we can go with 3000 or 3500. What do you think?

Reply:
Can we do 2500? I think that will be a nice time, considering the average typing speed of a person.

Reply:
Done @PrajvalRaval ✅